src/components/editor/publish-survey.tsx

```diff
@@ -0,0 +1,69 @@
+import { Stack } from "@mui/material";
+import { Device, DeviceSurvey, DeviceSurveyStatus, Survey } from "../../generated/client";
+import PropertiesPanel from "./properties-panel";
+import { useState } from "react";
+import PublishDeviceInfo from "./publish-device-info";
+import DevicesPanel from "./devices-panel";
+import PublishProperties from "./publish-properties";
+
+/**
+ * Components properties
+ */
+interface Props {
+  survey: Survey;
+  devices: Device[];
+  deviceSurveys: DeviceSurvey[];
+  publishSurveys: (deviceSurveys: DeviceSurvey[]) => Promise<void>;
+}
+
+/**
+ * Publish survey component
+ */
+const PublishSurvey = ({ survey, devices, deviceSurveys, publishSurveys }: Props) => {
+  const [selectedDevices, setSelectedDevices] = useState<Device[]>([]);
+
+  /**
+   * Publishes survey
+   *
+   * @param publishStartTime publish start time
+   * @param publishEndTime publish end time
+   */
+  const handlePublish = async (publishStartTime?: Date, publishEndTime?: Date) => {
+    if (!survey.id) return;
+    const deviceSurveys: DeviceSurvey[] = selectedDevices.map((device) => ({
+      deviceId: device.id!,
+      surveyId: survey.id!,
```

Comment:
```suggestion
      surveyId: survey.id,
```

Reply:
Typescript doesn't seem to realize that it cannot be null so build fails without the non-null assertion.

Reply:
Mysterious, well let's leave it as it is then